File: app/src/main/java/org/scp/gymlog/room/Converters.java
package org.scp.gymlog.room;

import androidx.room.TypeConverter;

import org.scp.gymlog.model.WeightSpecification;

import java.util.Calendar;
import java.util.Date;

public class Converters {
    @TypeConverter
    public static Calendar toDate(Long dateLong){
        if (dateLong == null) {
            return null;
        }
        Calendar cal = Calendar.getInstance();
        cal.setTimeInMillis(dateLong);
        return cal;
    }

    @TypeConverter
    public static Long fromDate(Calendar cal){
        return cal == null ? null : cal.getTimeInMillis();
    }

    @TypeConverter
    public static short fromWeightSpecification(WeightSpecification weightSpecification){
        return (short) weightSpecification.ordinal();
    }

    @TypeConverter
    public static WeightSpecification toWeightSpecification(short weightSpecification){
        return WeightSpecification.values()[weightSpecification];
    }
}

File: app/src/main/java/org/scp/gymlog/room/Converters.kt
package org.scp.gymlog.room

import androidx.room.TypeConverter
import org.scp.gymlog.model.ExerciseType
import org.scp.gymlog.model.WeightSpecification
import org.scp.gymlog.util.DateUtils.timeInMillis
import org.scp.gymlog.util.DateUtils.toLocalDateTime
import java.time.LocalDateTime

object Converters {
    @TypeConverter
    fun toDate(millis: Long?): LocalDateTime? {
        return millis?.toLocalDateTime
    }

    @TypeConverter
    fun fromDate(localDate: LocalDateTime?): Long? {
        return localDate?.timeInMillis
    }

    @TypeConverter
    fun fromWeightSpecification(weightSpecification: WeightSpecification): Short {
        return weightSpecification.ordinal.toShort()
    }

    @TypeConverter
    fun toWeightSpecification(weightSpecification: Short): WeightSpecification {
        return WeightSpecification.values()[weightSpecification.toInt()]
    }

    @TypeConverter
    fun fromExerciseType(exerciseType: ExerciseType): Short {
        return exerciseType.ordinal.toShort()
    }

    @TypeConverter
    fun toExerciseType(exerciseType: Short): ExerciseType {
        return ExerciseType.values()[exerciseType.toInt()]
    }
}